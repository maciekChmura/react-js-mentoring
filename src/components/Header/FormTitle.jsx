import React from 'react';
import styled from 'styled-components';

const H2 = styled.h2`
  grid-area: 2 / 2 / 3 / 3;
  font-size: 18px;
  color: white;
  width: auto;
  height: auto;
`;

const FormTitle = () => <H2>FIND YOUR MOVIE</H2>;

export default FormTitle;
