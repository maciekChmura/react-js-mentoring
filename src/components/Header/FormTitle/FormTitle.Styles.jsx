import styled from 'styled-components';

const H2 = styled.h2`
  grid-area: 2 / 2 / 3 / 3;
  font-size: 18px;
  color: white;
  width: auto;
  height: auto;
`;

H2.displayName = 'H2';

export default H2;
