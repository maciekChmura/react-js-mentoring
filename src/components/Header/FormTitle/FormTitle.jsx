import React from 'react';
import H2 from './FormTitle.Styles';

const FormTitle = ({ title }) => <H2>{title}</H2>;

export default FormTitle;
