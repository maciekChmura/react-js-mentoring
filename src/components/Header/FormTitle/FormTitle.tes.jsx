import React from 'react';
import renderer from 'react-test-renderer';
import FormTitle from './FormTitle';

test('FormTitle renders', () => {
  const component = renderer.create(<FormTitle />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
