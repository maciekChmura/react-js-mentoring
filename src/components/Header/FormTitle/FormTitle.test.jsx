import React from 'react';
import { shallow } from 'enzyme';
import FormTitle from './FormTitle';

describe('FormTitle', () => {
  it('should render correctly', () => {
    const component = shallow(<FormTitle />);

    expect(component).toMatchSnapshot();
  });

  it('should render props', () => {
    const component = shallow(<FormTitle title="test" />);

    // expect(component.find('H2').).toBe('test');
    expect(component.text()).toEqual('test');
  });
});
