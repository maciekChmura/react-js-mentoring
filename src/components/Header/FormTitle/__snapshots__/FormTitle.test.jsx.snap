// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTitle should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormTitle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "FormTitleStyles__H2-sc-10gmsuy-0",
        "isStatic": true,
        "rules": Array [
          "grid-area:2 / 2 / 3 / 3;font-size:18px;color:white;width:auto;height:auto;",
        ],
      },
      "displayName": "H2",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "FormTitleStyles__H2-sc-10gmsuy-0",
      "target": "h2",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "FormTitleStyles__H2-sc-10gmsuy-0",
          "isStatic": true,
          "rules": Array [
            "grid-area:2 / 2 / 3 / 3;font-size:18px;color:white;width:auto;height:auto;",
          ],
        },
        "displayName": "H2",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "FormTitleStyles__H2-sc-10gmsuy-0",
        "target": "h2",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
